packages/cashier/src/stores/payment-agent-store.ts

```diff
@@ -129,17 +143,13 @@ export default class PaymentAgentStore {
     }
 
     @action.bound
-    filterPaymentAgentList(bank) {
-        if (bank) {
-            this.filtered_list = [];
+    filterPaymentAgentList(bank?: number | string): void {
+        if (bank && typeof bank === 'string') {
+            this.filtered_list = [] as unknown as IObservableArray<TPartialPaymentAgentList>;
             this.list.forEach(payment_agent => {
                 const supported_banks = payment_agent?.supported_banks;
                 if (supported_banks) {
-                    const is_string = typeof supported_banks === 'string';
```

Comment:
remove because `supported_banks` is always an array now (`supported_banks : [{supported_bank: 'visa'}]`)